apps/api/scripts/delete-competitions.ts

```diff
@@ -0,0 +1,570 @@
+        .returning({ id: competitionsLeaderboard.id });
+      deletions.push({
+        table: "competitions_leaderboard",
+        count: leaderboardDeleted.length,
+      });
+
+      // Delete from competition_agents
+      const agentsDeleted = await tx
+        .delete(competitionAgents)
+        .where(eq(competitionAgents.competitionId, competitionId))
+        .returning({ agentId: competitionAgents.agentId });
+      deletions.push({
+        table: "competition_agents",
+        count: agentsDeleted.length,
+      });
+
+      // Delete from trading_competitions
+      const tradingCompDeleted = await tx
+        .delete(tradingCompetitions)
+        .where(eq(tradingCompetitions.competitionId, competitionId))
+        .returning({ id: tradingCompetitions.competitionId });
+      deletions.push({
+        table: "trading_competitions",
+        count: tradingCompDeleted.length,
+      });
+
+      // Finally, delete from competitions
+      const compDeleted = await tx
+        .delete(competitions)
+        .where(eq(competitions.id, competitionId))
+        .returning({ id: competitions.id });
+      deletions.push({ table: "competitions", count: compDeleted.length });
+
+      if (dryRun) {
+        // Rollback transaction in dry run mode
+        throw new Error("DRY_RUN_ROLLBACK");
+      }
+    });
+  } catch (error: unknown) {
+    if (error instanceof Error && error.message === "DRY_RUN_ROLLBACK") {
+      console.log(chalk.yellow("[DRY RUN] Transaction rolled back"));
+    } else {
+      throw error;
+    }
+  }
+
+  // Log deletion summary
+  console.log(
+    chalk.green(`\nDeletion summary for competition ${competitionId}:`),
+  );
+  deletions.forEach(({ table, count }) => {
+    if (count > 0) {
+      console.log(chalk.gray(`  - ${table}: ${count} rows`));
+    }
+  });
+
+  return deletions;
+}
+
+/**
+ * Recalculate agent scores after deleting ended competitions
+ * @param excludedCompetitionIds - Competition IDs to exclude from rank calculations
+ */
+async function recalculateAgentScores(excludedCompetitionIds: string[] = []) {
+  console.log(chalk.blue("\nRecalculating agent scores from history..."));
+
+  if (excludedCompetitionIds.length > 0) {
+    console.log(
+      chalk.yellow(
+        `Excluding competitions from rank calculations: ${excludedCompetitionIds.join(", ")}`,
+      ),
+    );
+  }
+
+  try {
+    // First, if we have excluded competitions, delete their history entries
+    if (excludedCompetitionIds.length > 0) {
+      const deletedHistory = await db
+        .delete(agentScoreHistory)
+        .where(inArray(agentScoreHistory.competitionId, excludedCompetitionIds))
+        .returning({ id: agentScoreHistory.id });
+
+      console.log(
+        chalk.gray(
+          `Deleted ${deletedHistory.length} history entries for excluded competitions`,
+        ),
+      );
+    }
+
+    // Get the most recent score history for each agent
+    const latestScoresRaw = await db.execute(sql`
```

Comment:
i reworked this logic and compared the script's output against a dump from july 17th. the number of `agent_score` entries should be 38, and i wasnt getting that with the previous logic.

instead of recalculating the `agent_score` values...we can just get the cached value from `agent_score_history`.